sql/item_windowfunc.h

```diff
@@ -683,6 +686,7 @@ class Item_sum_ntile : public Item_sum_window_with_row_count
  private:
   longlong get_num_quantiles() { return args[0]->val_int(); }
   ulong current_row_count_;
+  longlong n_old_val_;
```

Comment:
Why `longlong`, can it ever be negative?
I would prefer here `ulonglong`.

Reply:
Then we not allowed to compare num_quantiles != n_old_val_

Reply:
Then solve it with casting.

Reply:
I've used reinterpret_cast <longlong> (n_old_val_) 
Does it look fine?

Reply:
May I use 
auto n_old_val_   
with c++11 set(CMAKE_CXX_STANDARD 11)

Reply:
I've used static_cast.  Does it look fine?